apps/opik-backend/src/main/java/com/comet/opik/infrastructure/bi/UsageReportDAO.java

```diff
@@ -0,0 +1,80 @@
+package com.comet.opik.infrastructure.bi;
+
+import com.google.inject.ImplementedBy;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.UnableToExecuteStatementException;
+
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.Optional;
+
+@ImplementedBy(UsageReportDAOImpl.class)
+interface UsageReportDAO {
+
+    Optional<String> getAnonymousId();
+
+    void saveAnonymousId(@NonNull String id);
+
+    boolean isEventReported(@NonNull String eventType);
+
+    void addEvent(@NonNull String eventType);
+
+    void markEventAsReported(@NonNull String eventType);
+}
+
+@Slf4j
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+@Singleton
+class UsageReportDAOImpl implements UsageReportDAO {
```

Comment:
There's no good separation of concerns in this DAO as it handles both `metadata` and `usage_information` model. It should be separated into two.

Reply:
I can split it; I just thought it was not worth it at the moment since it's used for a single purpose.